gpcontrib/arenadata_toolkit/src/arenadata_toolkit_worker.c

```diff
@@ -0,0 +1,450 @@
+#include "postgres.h"
+
+#include "access/xact.h"
+#include "access/genam.h"
+#include "access/heapam.h"
+#include "catalog/pg_db_role_setting.h"
+#include "catalog/pg_extension.h"
+#include "catalog/indexing.h"
+#include "cdb/cdbdisp_query.h"
+#include "cdb/cdbdispatchresult.h"
+#include "libpq-fe.h"
+#include "postmaster/bgworker.h"
+#include "storage/proc.h"
+#include "storage/ipc.h"
+#include "utils/snapmgr.h"
+#include "utils/builtins.h"
+#include "utils/fmgroids.h"
+
+#include "arenadata_toolkit_worker.h"
+#include "arenadata_toolkit_guc.h"
+#include "tf_shmem.h"
+
+#define EXTENSIONNAME "arenadata_toolkit"
+
+typedef struct
+{
+	Oid			dbid;
+	bool		get_full_snapshot_on_recovery;
+}	tracked_db_t;
+
+static BackgroundWorker worker;
```

Comment:
why not to move the static declaration inside the `arenadata_toolkit_worker_register()`?